doc/src/contribute/observability.md

```diff
@@ -210,26 +210,39 @@ To see nested spans visualized with [Jaeger](https://www.jaegertracing.io), do t
 
 ### Memory profiling
 
-jemalloc has a memory profiling mode that can be enabled at runtime with environment variables. See:
+Sui uses the jemalloc memory allocator by default on most platforms, and there is code that enables automatic
+memory profiling using jemalloc's sampling profiler, which is very lightweight and designed for production
+use.  The profiling code spits out profiles at most every 5 minutes, and only when total memory has increased
+by a default 20%.  Profiling files are named `jeprof.<TIMESTAMP>.<memorysize>MB.prof` so that it is easy to 
+correlate to metrics and incidents, for ease of debugging.
 
-* https://github.com/jemalloc/jemalloc/wiki/Use-Case%3A-Heap-Profiling
-* https://gist.github.com/ordian/928dc2bd45022cddd547528f64db9174
+For the memory profiling to work, the following environment variables need to be set (and if you use the 
+Docker image, they are set automatically):
 
-For example, set `JE_MALLOC_CONF` or `JEMALLOC_SYS_WITH_MALLOC_CONF` to:
-`prof:true,lg_prof_interval:24,lg_prof_sample:19`
+* `MALLOC_CONF=prof:true`
+* `_RJEM_MALLOC_CONF=prof:true`
 
-The above setting means: turn on profiling, sample every 2^19 or 512KB bytes allocated,
-and dump out the profile every 2^24 or 16MB of memory allocated.
+Note that running some allocator-based heap profilers such as [Bytehound](https://github.com/koute/bytehound) will essentially disable
+automatic jemalloc profiling, because they interfere with or don't implement `jemalloc_ctl` stats APIs.
 
 To view the profile files:
 1. `brew install jemalloc libunwind gprof2dot`
```

Comment:
Yes good point.  I believe I was able to run jeprof on Mac, but it doesn't matter because one needs the same platform as where one is going to debug.   These instructions are old because we have a Docker image with debugging tools, but the problem is that that Docker image is not public yet.  I'll update the instructions though to not have brew install, maybe I'll put in the Debian instructions